src/datasources/asset/AssetDataSource.ts

```diff
@@ -82,4 +84,11 @@ export class AssetDataSource extends DataSourceBase<AssetQuery, AssetDataSourceO
   getListAssetsSource(): ListAssetsDataSource {
     return this.listAssetsDataSource;
   }
+
+  async metricFindQuery(query: AssetQuery): Promise<MetricFindValue[]> {
+    console.log(query);
+    const assetFilter = query as ListAssetsQuery;
+    const assetsResponse: AssetModel[] = await this.getListAssetsSource().queryAssets(assetFilter?.filter ?? '', 1000);
```

Comment:
`this.getListAssetsSource().queryAssets(assetFilter, QUERY_LIMIT);`

We could define a constant for 1k as it's the upper limit we allow.